### PR TITLE
[POC] setting up tests with eos on drone-ci

### PR DESCRIPTION
**Pull request by dpakach**
this PR adds scripts for creating a new eos instance in HCloud so that we can run our API tests with the EOS storage backend.

### CODE REVIEW

File: tests/spwan_eos.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+ME=$DRONE_HCLOUD_USER
+SERVER_NAME=dpakach-eos-ocis-test # droneci-eos-test-${DRONE_COMMIT_ID}
+
+# Create a new machine on hcloud for eos
+hcloud server create --type cx21 --image ubuntu-20.04 --ssh-key $ME --name $SERVER_NAME --label owner=$ME --label for=test --label from=eos-compose
+
+IPADDR=$(hcloud server ip $SERVER_NAME)
+OCIS_DOMAIN=$(hcloud server ip $SERVER_NAME)
```
Comment (ownclouders) [inline tests/spwan_eos.sh:9]: Codacy found an issue: [OCIS_DOMAIN appears unused. Verify use (or export if used externally).](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6117463)

Comment (ownclouders) [inline tests/spwan_eos.sh:9]: Codacy found an issue: [OCIS_DOMAIN appears unused. Verify use (or export if used externally).](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6117463)

File: tests/spwan_eos.sh [hunk 2]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+ME=$DRONE_HCLOUD_USER
+SERVER_NAME=droneci-eos-test-${DRONE_COMMIT_ID}
+
+# Create a new machine on hcloud for eos
+hcloud server create --type cx21 --image ubuntu-20.04 --ssh-key $ME --name $SERVER_NAME --label owner=$ME --label for=test --label from=eos-compose
+
+IPADDR=$(hcloud server ip $SERVER_NAME)
+OCIS_DOMAIN=$(hcloud server ip $SERVER_NAME)
```
Comment (ownclouders) [inline tests/spwan_eos.sh:9]: Codacy found an issue: [OCIS_DOMAIN appears unused. Verify use (or export if used externally).](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6117463)

**Comment by dpakach** [comment]
@felixboehm , I have added some scripts that could start up a new hcloud machine with eos.
If we setup the hcloud env with `HCLOUD_TOKEN` and then run the `tests/spwan_eos.sh` which should be able to start an eos machine.
We will just have to provide `DRONE_HCLOUD_USER` and `DRONE_COMMIT_ID`

**Comment by felixboehm** [comment]
Looks good!
Did you run the scripts for setup, or did you test same steps manually??
Setup works reliably??

File: tests/spwan_eos.sh [hunk 3]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
```
Comment (PVince81) [inline tests/spwan_eos.sh:1]: file name s/spwan/spawn/

**Comment by dpakach** [comment]
> Looks good!
> Did you run the scripts for setup, or did you test same steps manually??
> Setup works reliably??

I ran the scripts from my local machine and it was able to create a new machine with eos setup on hcloud. If we add credentials for hcloud in drone, it should be able to do the same.

File: tests/spawn_eos.sh [hunk 4]
```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+ME=$DRONE_HCLOUD_USER
```
Comment (ownclouders) [inline tests/spawn_eos.sh:2]: Codacy found an issue: [ME appears unused. Verify use (or export if used externally).](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6117463)

File: tests/spawn_eos.sh [hunk 5]
```diff
@@ -0,0 +1,35 @@
...
+SERVER_NAME=droneci-eos-test-${DRONE_COMMIT_ID}-${RUN_PART}
```
Comment (ownclouders) [inline tests/spawn_eos.sh:3]: Codacy found an issue: [SERVER_NAME appears unused. Verify use (or export if used externally).](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6117463)

File: tests/spawn_eos.sh (hunk 4 above)
Comment (ownclouders) [inline tests/spawn_eos.sh:2]: Codacy found an issue: [ME appears unused. Verify use (or export if used externally).](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6117463)

File: tests/spawn_eos.sh (hunk 5 above)
Comment (ownclouders) [inline tests/spawn_eos.sh:3]: Codacy found an issue: [SERVER_NAME appears unused. Verify use (or export if used externally).](https://app.codacy.com/gh/owncloud/ocis/pullRequest?prid=6117463)

**Comment by dpakach** [comment]
Blocked by https://github.com/owncloud/ocis/issues/586

**Comment by butonic** [comment]
eos tests are no longer covered by ocis ci. it should run in reva or elsewhere.

**Pull request closed without merging**